operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -588,14 +589,22 @@ private static String formatHashLabel(String hash) {
   protected V1PodSpec createSpec(TuningParameters tuningParameters) {
     V1PodSpec podSpec = createPodSpec(tuningParameters)
         .readinessGates(getReadinessGates())
-        .initContainers(getServerSpec().getInitContainers());
+        .initContainers(getServerSpec().getInitContainers().stream()
+                .map(c -> c.env(createEnv(c, tuningParameters))).collect(Collectors.toList()));
 
     for (V1Volume additionalVolume : getVolumes(getDomainUid())) {
       podSpec.addVolumesItem(additionalVolume);
     }
     return podSpec;
   }
 
+  private List<V1EnvVar> createEnv(V1Container c, TuningParameters tuningParameters) {
+    List<V1EnvVar> envVars = new ArrayList<>();
+    Optional.ofNullable(c.getEnv()).ifPresent(envVars::addAll);
```

Comment:
should env var configued in initContainer take precedence over those configured in server or cluster?

Reply:
That's a good point. I had thought that the env var list will be mutually exclusive or env variables configured in server or cluster will take precedence. I'm not sure if customers would want to override  server/domain env variables in the init containers. @rjeberhard thoughts? Thanks.

Reply:
I guess that I would lean the other direction... That is, the highest precedence would be the most local configuration, which in this case, I think would be the environment variables specifically on the initContainer.

Reply:
I have made the change to set higher precedence for environment variables set in init containers in cad0c30. Please let me know if I missed something. Thanks.

Reply:
please add a test case for testing the precedence. Thanks

Reply:
Added `whenInitContainersHaveEnvVar_verifyInitContainersEnvVarTakesPrecedenceOverPreConfiguredEnvVar` test in 66f56c5